test/controllers/products_controller_test.rb

```diff
@@ -26,13 +26,66 @@
     end
   end
 
-  it "must get the show view" do
-    get product_path(one.id)
-    must_respond_with :success
+  describe "show" do
+    it "must get the show view" do
+      get product_path(one.id)
+      must_respond_with :success
+
+      get product_path(-1)
+      must_respond_with :redirect
+      must_redirect_to root_path
+    end
+  end
+
+  describe "new product" do
+    it "should get new if a mercant is signed in" do
+      @merchant = merchants(:sappy1)
+      login(@merchant, :github)
+      get new_product_path
+      must_respond_with :success
+    end
+
+    it "must redirect to root path if a merchant is not signed in" do
+      # With no mercant signed in
+      get edit_product_path(one.id)
+      must_respond_with :redirect
+      flash[:status].must_equal :error
+      must_redirect_to root_path
+    end
+  end
+
+  describe "create product" do
+    before do
+      @merchant = merchants(:sappy1)
+      login(@merchant, :github)
+    end
+
+    it "should be able to successfully create a new product" do
+        proc {
+          post products_path, params: { name: "newbie", price: 2, category: "new category", quantity: 1, merchant_id: @merchant.id }
+        }.must_change 'Product.count', 1
+      must_respond_with :redirect
```

Comment:
the create method will respond with redirect if the product failed to save, as well. might be a good idea to add a flash[:status] and flash[:result_text] that can be tested as well. but this is definitely sufficient since you're testing that the Product.count increased by 1.